src/Cake.NuGet/NuGetContentResolver.cs

```diff
@@ -72,6 +76,11 @@ private IReadOnlyCollection<IFile> GetAddinAssemblies(DirectoryPath path, Packag
                 // Get relative path.
                 var relative = path.GetRelativePath(assembly.Path);
                 var framework = ParseFromDirectoryPath(current, relative.GetDirectory());
+                if (framework == null)
+                {
+                    _log.Warning("Unable to parse compatible framework from path {0}. Setting any framework.", assembly.Path.MakeAbsolute(_environment).FullPath);
```

Comment:
In this particular context, saying `falling back to using root folder of NuGet package.` is not the truth. When we can't parse framework from the path (be it `foo/bar/lib.dll` or just `lib.dll` we will treat it as compatible by setting framework to `AnyFramework`. By setting that, we can try loading it if we can't find a better match (e.g. `lib/net45/...`, `lib/net46/...`)